codebasin/config.py

```diff
@@ -151,6 +156,88 @@ def __call__(
             dest.extend(matches)
 
 
+@dataclass
+class _CompilerMode:
+    name: str
+    defines: list[str] = field(default_factory=list)
+    include_paths: list[str] = field(default_factory=list)
+    include_files: list[str] = field(default_factory=list)
+
+    @classmethod
+    def from_toml(cls, toml: object) -> Self:
+        return _CompilerMode(**toml)
+
+
+@dataclass
+class _CompilerPass:
+    name: str
+    defines: list[str] = field(default_factory=list)
+    include_paths: list[str] = field(default_factory=list)
+    include_files: list[str] = field(default_factory=list)
+    modes: list[str] = field(default_factory=list)
+
+    @classmethod
+    def from_toml(cls, toml: object) -> Self:
+        return _CompilerPass(**toml)
+
+
+@dataclass
+class _Compiler:
```

Comment:
We have may discussed this, but is there a reason why you want to keep `_Compiler` private?

Is the intention to not expose `_Compiler` to a **user**, but still accessible for developers?

Reply:
As discussed offline, the intent is to keep `_Compiler` private for the time being so that we don't have to worry about introducing breaking changes. `ArgumentParser` is public, but there's no need to expose the guts (yet).